org.eclipse.xtext.ide.tests/templates/templates.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"
+  encoding="UTF-8"
+  standalone="no"?>
+
+<templates>
+  <template
+    context="org.eclipse.xtext.ide.tests.testlanguage.TestLanguage.TypeDeclaration"
+    description="Add type declaration"
+    name="type declaration template">type ${1:name} {
+  $0
```

Comment:
For my understanding: Does this template work in Eclipse, too? Because there I'd expect something like $cursor instead of $0

Reply:
I don't believe the syntax is the same, so I wouldn't expect it to work. These templates are based on the LSP snippet syntax https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#snippet_syntax

Reply:
so in xtext templates there is stuff like
`${Name}` `${public:Enum('Visibility')}` `${event:CrossReference('Transition.event')}`
i assume the first one you can easily adapt to lsp variant.
for the others i assume it needs more compilated calculation to achieve something like
https://github.com/itemis/xtext-languageserver-example/blob/dfbb365d70e1370db4d4b60bc193efa173082266/org.xtext.example.mydsl.ide/src/org/xtext/example/mydsl/ide/contentassist/MyDslIdeContentProposalProvider.java#L28

Reply:
In the LSP snippet syntax, tab-stops are numbered, so ${Name} would become ${1:Name}. The others use JFace resolvers which I have assumed are not available on a server, so are not supported in this commit.

Reply:
So yes, something more complicated is required to convert an enum resolver to a snippet choice.

Reply:
To be honest, I'm not really a fan of the idea of duplicating the template definitions between LSP and Eclipse. I'd rather like to see templates being reused. Would you agree that the Eclipse templates could be translated into the LSP format while reading them from the XML file?

Reply:
For context, these are just the classes we have used for our project to get something working, and we thought they might be useful to you. If they go against your preferred implementation direction, then we will be fine to drop the PR.

To answer your question: Yes the xml files could be translated as they are read, but...

- The obvious issue is that there is a performance hit to translating the xml file each time and we didn't want to take that hit when the result of the translation could be written out to a new file.
- There are also variables that can be added to a snippet that are to be evaluated in the LSP client rather than the server - in a traditional Eclipse setup there would not be an LSP client to evaluate these.
- Also, the syntax for a client-side-variable-with-default-value is the same as a jface-variable-with-no-argument-resolver, so a translator would need to know which case it was trying to translate.